### PR TITLE
Simplify Storage samples and their tests.

### PR DESCRIPTION
**Pull request by jmdobry**
- [x] - Using ES2015 (string templates, `const`, `let`, and arrow functions
- [x] - Unit tests are now really small (they only test the error cases)
- [x] - System tests now exercise the CLI of the samples
- [x] - Achieved 100% code coverage for `files.js`, `buckets.js`, `acl.js`, and `encryption.js`
- [x] - Samples were updated to follow latest sample usability guidelines

I left the Storage Transfer API samples alone. I don't want to mess with them until the google-cloud-node has support for the Storage Transfer API.

### CODE REVIEW

File: storage/encryption.js [hunk 1]
```diff
@@ -39,131 +36,120 @@ var crypto = require('crypto');
...
-  .example('node $0 upload my-bucket resources/test.txt file_encrypted.txt QxhqaZEqBGVTW55HhQw9Q=', 'Upload "resources/test.txt" to "gs://my-bucket/file_encrypted.txt".')
-  .example('node $0 download my-bucket file_encrypted.txt ./file.txt QxhqaZEqBGVTW55HhQw9Q=', 'Download "gs://my-bucket/file_encrypted.txt" to "./file.txt".')
-  .example('node $0 rotate my-bucket file_encrypted.txt QxhqaZEqBGVTW55HhQw9Q= SxafpsdfSDFS89sds9Q=', 'Rotate encryptiong keys for "gs://my-bucket/file_encrypted.txt".')
+  .example(`node $0 generate-encryption-key`, `Generate a sample encryption key.`)
+  .example(`node $0 upload my-bucket ./resources/test.txt file_encrypted.txt QxhqaZEqBGVTW55HhQw9Q=`, `Encrypts and uploads "resources/test.txt" to "gs://my-bucket/file_encrypted.txt".`)
+  .example(`node $0 download my-bucket file_encrypted.txt ./file.txt QxhqaZEqBGVTW55HhQw9Q=`, `Decrypts and downloads "gs://my-bucket/file_encrypted.txt" to "./file.txt".`)
+  .example(`node $0 rotate my-bucket file_encrypted.txt QxhqaZEqBGVTW55HhQw9Q= SxafpsdfSDFS89sds9Q=`, `Rotates encryptiong keys for "gs://my-bucket/file_encrypted.txt".`)
```
Comment (ace-n) [inline storage/encryption.js]: Typo: encryption**g**.

File: storage/system-test/acl.test.js [hunk 2]
```diff
@@ -13,225 +13,92 @@
...
-  after(function (done) {
-    storage.bucket(bucketName).deleteFiles({ force: true }, function (err) {
-      if (err) {
-        return done(err);
-      }
-      storage.bucket(bucketName).delete(done);
+  after((done) => {
+    storage.bucket(bucketName).file(fileName).delete((err) => {
+      assert.ifError(err);
+      setTimeout(() => storage.bucket(bucketName).delete(done), 2000);
```
Comment (ace-n) [inline storage/system-test/acl.test.js]: I take it you want this to return an error if the bucket `delete()` operation fails?

File: storage/acl.js [hunk 3]
```diff
@@ -175,45 +297,49 @@ var program = module.exports = {
 
 cli
   .demand(1)
-  .command('add <entity> <role>', 'Add access controls on a bucket or file.', {}, function (options) {
-    program.addAccessControl(utils.pick(options, ['entity', 'role', 'bucket', 'default', 'file']), utils.makeHandler());
+  .command(`print-bucket-acl <bucketName>`, `Prints the ACL for a bucket.`, {}, (opts) => {
```
Comment (jmdobry) [inline storage/acl.js]: I copied this from the python sample. I figure it was better to be consistent.

File: storage/acl.js [hunk 4]
```diff
@@ -11,162 +11,284 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
```
Comment (jmdobry) [inline storage/acl.js]: Done

File: storage/encryption.js (hunk 1 above)
Comment (jmdobry) [inline storage/encryption.js]: Fixed

File: storage/system-test/acl.test.js (hunk 2 above)
Comment (jmdobry) [inline storage/system-test/acl.test.js]: Fixed

**Pull request merged**